lib/util/http/index.ts

```diff
@@ -57,6 +57,40 @@ async function resolveResponse<T>(
   }
 }
 
+// isAmazon return true if request options contains Amazon related headers
+function isAmazon(opts: any): boolean {
+  return opts.search?.includes('X-Amz-Algorithm');
+}
+
+// isAzureBlob return true if request options contains Azure container registry related data
+function isAzureBlob(opts: any): boolean {
+  return (
+    opts.hostname?.endsWith('blob.core.windows.net') &&
+    opts.href?.includes('/docker/registry')
+  ); // lgtm [js/incomplete-url-substring-sanitization]
+}
+
+// removeAuthorizationHeaders from the redirect options
+export function removeAuthorizationHeaders(opts: any): any {
```

Comment:
@rarkins I think it makes sense to move this to `util/http/auth.ts`. What do you think?

Reply:
I don't mind either way

Reply:
Ok, for me it looks cleaner to move it to auth. So @etiennetremel can you please move that to the suggested file. 🙃